server/task/src/main/scala/org/apache/james/task/eventsourcing/TerminationSubscriber.scala

```diff
@@ -0,0 +1,50 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.task.eventsourcing
+
+import java.util.concurrent.{LinkedBlockingQueue, TimeUnit}
+
+import org.apache.james.eventsourcing.{Event, Subscriber}
+import org.reactivestreams.Publisher
+
+import reactor.core.publisher.Flux
+
+trait TerminationSubscriber extends Subscriber {
+  override def handle(event: Event): Unit = event match {
+    case event: Completed => addEvent(event)
```

Comment:
I think it does not add more value.

Maybe it's worst: you modify your business logic for a technical usage.

Reply:
> I think it does not add more value.

> Maybe it's worst: you modify your business logic for a technical usage.

I think the opposite: we need this concept when we talk to each others or when we define tickets. It's a sign, to me, that it's a domain concept.

Further more, the pattern matching with a catch-all will hide the fact we don't handle potential new types. There are two solutions: either we list all types and avoid the catch-all entry (that would actually define what is a TerminationEvent and what is not) or we promote the concept in the type system because we think it's a domain thing.

WDYT?